### DISCUSSION TITLE
CIFAR10 FEDAVG SIMULATION ERROR

### DISCUSSION THREAD

**Discussion by FabioNotaro2001** [NVIDIA/NVFlare#1319] [Q&A] [answered]
Hi', I'm trying to execute the example of cifar10_fedavg present in this link: https://github.com/NVIDIA/NVFlare/blob/main/examples/cifar10/cifar10-sim/README.md

In particular, I put myself into the directory cifar10/cifar10-sim and I executed the commands 
python3 ../pt/utils/cifar10_download_data.py
export RESULT_ROOT=/tmp/nvflare/sim_cifar10
./set_alpha.sh cifar10_fedavg 0.5
nvflare simulator job_configs/cifar10_fedavg --workspace ${RESULT_ROOT}/fedavg_alpha0.5 --threads 8 --n_clients 8

But after this last command, I expected the training to start, instead I got this long sequence of errors:

![Schermata del 2023-02-08 16-07-18](https://user-images.githubusercontent.com/93600518/217571204-9b4e9acf-4e70-4d9d-a5d3-0c38c01b612d.png)
![Schermata del 2023-02-08 16-07-40](https://user-images.githubusercontent.com/93600518/217571227-5c007767-da6a-49c7-ae40-cdd7b95b941e.png)
![Schermata del 2023-02-08 16-07-51](https://user-images.githubusercontent.com/93600518/217571233-76a68af0-d3c5-4941-8d74-a21296fba11d.png)

**Comment by YuanTingHsieh** [comment, marked as answer]
Hello, thanks for the interest.

The problem is "No module named 'pt'".
Did you do "export PYTHONPATH=${PWD}/.."?

Can you try to check if your PYTHONPATH include its parent?

**Reply by FabioNotaro2001** [reply]
Yes , now it works! Thank you  very much. One last question: why this command isn't inserted in the simulator example with hello-pt-tb and instead here it is necessary?

**Reply by YuanTingHsieh** [reply]
good question, we will try to standardize example structure in the future.

But now they are a bit different. 

So if you compare the configurations between these two examples, you will find that cifar10 needs codes from its "pt" folder.
Which is this folder: https://github.com/NVIDIA/NVFlare/tree/main/examples/cifar10/pt
So that's why it requires to add this path.
The reason it has this "pt" folder outside is because it wants to reuse the same code for all cifar10 jobs.

VS

hello-pt-tb example we only use things in "custom" folder, and NVFlare add codes in "custom" folder automatically.
